src/theory/builtin/theory_builtin_rewriter.cpp

```diff
@@ -214,26 +213,57 @@ Node TheoryBuiltinRewriter::getArrayRepresentationForLambdaRec(TNode n,
   std::vector< Node > conds;
   std::vector< Node > vals;
   Node curr = n[1];
-  while( curr.getKind()==kind::ITE || curr.getKind()==kind::EQUAL || curr.getKind()==kind::NOT ){
-    Trace("builtin-rewrite-debug2") << "  process condition : " << curr[0] << std::endl;
+  Kind ck = curr.getKind();
+  NodeManager* nm = NodeManager::currentNM();
```

Comment:
Nit: you could move this to beginning of the method so that you use `nm` also in the if block before this.